sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -361,10 +361,18 @@ case class JsonTuple(children: Seq[Expression])
   // the fields to query are the remaining children
   @transient private lazy val fieldExpressions: Seq[Expression] = children.tail
 
+  // a field name given with constant null will be replaced with this pseudo field name
+  private val nullFieldName = "__NullFieldName"
```

Comment:
Yeah, I've also considered using Option here. But don't want to come out Option version first, so we can experience review process. It looks good to me.